kart/dataset_mixins.py

```diff
@@ -46,3 +50,130 @@ def diff_to_wc(self, repo, ds_filter=DatasetKeyFilter.MATCH_ALL):
         """
         # Subclasses to override.
         pass
+
+    def get_raw_diff_for_subtree(self, other, subtree_name, reverse=False):
+        """
+        Get a pygit2.Diff of the diff between some subtree of this dataset, and the same subtree of another dataset
+        (generally the "same" dataset at a different revision).
+        """
+
+        flags = pygit2.GIT_DIFF_SKIP_BINARY_CHECK
```

Comment:
should we be using `GIT_DIFF_FORCE_BINARY` here?

> `GIT_DIFF_FORCE_BINARY`
> 
> Treat all files as binary, disabling text diffs
>
> `GIT_DIFF_SKIP_BINARY_CHECK`
> 
> Disable updating of the binary flag in delta records. This is useful when iterating over a diff if you don't need hunk and data callbacks and want to avoid having to load file completely.
>
> ([source](https://libgit2.org/libgit2/#HEAD/type/git_diff_option_t))